### PR TITLE
Travis setup and Dockerfile clean up

### PR DESCRIPTION
**Pull request by AbdealiLoKo**

### CODE REVIEW

File: Dockerfile.ubuntu [hunk 1]
```diff
@@ -60,6 +49,3 @@ RUN pip install -r /opt/file-metadata/test-requirements.txt
 
 # Install package
 RUN pip install /opt/file-metadata --upgrade
-
```
Comment (drtrigon) [inline Dockerfile.ubuntu]: We need some command to run in the end to test the successful install. At least an import test.

File: Dockerfile.centos [hunk 2]
```diff
@@ -74,6 +65,3 @@ RUN pip install -U cython
 
 # Install package
 RUN pip install /opt/file-metadata --upgrade
-
```
Comment (drtrigon) [inline Dockerfile.centos]: We need some command to run in the end to test the successful install. At least an import test.

File: Dockerfile.centos [hunk 3]
```diff
@@ -74,6 +65,3 @@ RUN pip install -U cython
...
-# Test file-metadata import
-RUN python -c'import file_metadata; print(file_metadata.__version__)'
```
Comment (drtrigon) [inline Dockerfile.centos:79]: We need some command to run in the end to test the successful install. At least an import test.

File: Dockerfile.ubuntu [hunk 4]
```diff
@@ -60,6 +49,3 @@ RUN pip install -r /opt/file-metadata/test-requirements.txt
...
-# Test file-metadata import
-RUN python -c'import file_metadata; print(file_metadata.__version__)'
```
Comment (drtrigon) [inline Dockerfile.ubuntu:65]: We need some command to run in the end to test the successful install. At least an import test.

**Comment by drtrigon** [comment]
I cannot comment on the travis yet.

File: Dockerfile.ubuntu (hunk 4 above)
Comment (AbdealiLoKo) [inline Dockerfile.ubuntu:65]: I don't see why - Travis is already going to be testing the docker image, and all testing would happen on travis. This adds an additional commit that users would have to download

Comment (drtrigon) [inline Dockerfile.ubuntu:65]: The question is not whether file-metadata install works (that does travis - agreed), the question is whether the Dockerfile is valid! Imagine you forgot to add a package in Dockerfile (e.g. libmagickwand-dev).

Comment (AbdealiLoKo) [inline Dockerfile.ubuntu:65]: It seems there is still some confusion as to what travis does. Travis runs the unittests inside the docker image. So, if the libmagickwand-dev is not there, some of the travis unittests will fail.
Printing the `__version__` will not fail even if the libmagickwand-dev is not installed (Although pip install will fail)

File: Dockerfile.centos [hunk 5]
```diff
@@ -49,13 +40,13 @@ RUN yum install -y ffmpeg
...
 # Dlib
-RUN yum install -y cmake boost-python python-devel boost boost-devel gcc-c++ openblas-devel liblapack-devel
+RUN yum install -y cmake boost-python python-devel boost boost-devel gcc-gfortran gcc-c++ openblas-devel liblapack-devel
 
 # ZBar
 RUN yum install -y zbar-devel python-devel
 
 # Add the repository
-RUN git clone https://github.com/AbdealiJK/file-metadata.git /opt/file-metadata
+RUN git clone https://github.com/pywikibot/file-metadata.git /opt/file-metadata
```
Comment (drtrigon) [inline Dockerfile.centos:50]: `pywikibot` should be `pywikibot-catfiles`, right?

Comment (AbdealiLoKo) [inline Dockerfile.centos:50]: Yep, fixed now

**Comment by drtrigon** [comment]
Regarding the build issues on travis: Have you tried using my version of `Dockerfile.ubuntu` does it work then?
If I want to work on that branch: What is the easiest way to do so? Switch the default branch here on github temporarily?

**Comment by AbdealiLoKo** [comment]
Can you remove your last commit and merge this? Then your can create another branch from master.
Although tests fail, this seems like an improvement to master - hence merging even though tests fail should be ok

**Comment by drtrigon** [comment]
Sorry the last commit was an "midnight-5-mins-before-I-go-to-bed-desperate-try" - and a silly one... ;))
I reverted it locally. I'm working on the docker right now and prefer to investigate before I commit again.

**Comment by drtrigon** [comment]
build ok now: https://travis-ci.org/pywikibot-catfiles/docker-file-metadata/jobs/145637309

`Dockerfile.ubuntu` includes a "dirty" patch since I did not want to mess your code up... ;))
It's to `tests/image/image_file_test.py` and can be found in: https://gist.github.com/drtrigon/b7d367514745715fff332b652349f222

We should improve a bit on the testing debug output... `logging` module and pytest seam no to work very well together, I had to use the `-s` switch for pytest. I don't know what's the best solution but I guess the `-s` and an additional switch in out test scripts that allows to set the logging level from command line (e.g. using `logging.basicConfig(level=logging.WARN)`) would be nice. Another possibility could be to add `stream=sys.stdout` in the `basicConfig`.
Then maybe we should have `logging.warn` in front of each `return` for debugging, on the other hand `pycolorname` has a LOT of `logging.debug`.

The real issue was not existing cascade xml files and wrong paths to them in `def _haarcascade`. We must add either an `os.path.exists` and output `logging.warn` or have an explicit unittest that checks for their existence.
Also where do we store cascade files that do not come with cv? (e.g. some in catimages or self-trained ones) How do we install them?

Do we use `libav` in `Dockerfile.ubuntu` or should we install `ffmpeg` as well:
`sudo apt-get install ffmpeg`
and may be also the additional free (and non-free?) codecs:
`sudo apt-get install libavcodec-extra-53 libavdevice-extra-53 libavformat-extra-53 libavutil-extra-51 libpostproc-extra-52 libswscale-extra-2`

**Comment by drtrigon** [comment]
https://travis-ci.org/pywikibot-catfiles/docker-file-metadata/jobs/145671019
On my machine it had no issues, may be you know what's going on - otherwise I'll have another look later. The build time is quite close to 30 mins... btw.

Compared to the last comment above and the `ubuntu-14.04` case, here are the opencv cascade xml file paths are ok - no patch had to be applied. So this is clearly different to `ubuntu-14.04`.

Then I was wondering why are we using pip 7.0.0 to install file-metadata and all its deps here? Especially after explicitely using most recent pip at the top of the Dockerfile.

**Comment by AbdealiLoKo** [comment]
> Also where do we store cascade files that do not come with cv?

We install them with a separate class in setupdeps.py - Haven't needed them yet, so it's not created yet.

> Do we use libav in Dockerfile.ubuntu or should we install ffmpeg as well:

We use either libav OR ffmpeg - both aren't needed. ffmpeg doesnt exist in ubuntu 14.04 and 16.04's apt-get, hence we would use libav there.

> sudo apt-get install libavcodec-extra-53 libavdevice-extra-53 libavformat-extra-53 libavutil-extra-51 libpostproc-extra-52 libswscale-extra-2

Do these add more functionality to ffmpeg ? If yes, we should add unittests (that get skipped when they're not installed) to test whether filemetadata plays well with them !

**Comment by AbdealiLoKo** [comment]
I do not know why there's pip 7.0.0 either :D I thought you had added it :P 
Could we remove that and try ?

(I have not cleaned up the CentOS dockerfile completely ... I thought I should wait and see if my changes make sense on Ubuntu and can replicate on CentOS if they do make sense)

**Comment by drtrigon** [comment]
Removed the 7.0.0 version. And ran it with `-s` for pytest, here the result: http://dpaste.com/0RD8H1F

I assume it's most probably a issue with zxing: http://stackoverflow.com/questions/29864316/could-not-find-or-load-main-class-com-google-zxing-client-j2se-commandlinerunner

Do you know how to solve this?

**Pull request merged**